From da3d870bd99ad7b547c17fb5f398f6f599f14786 Mon Sep 17 00:00:00 2001
From: Jason Lowe <jlowe@apache.org>
Date: Fri, 8 May 2015 13:45:35 +0000
Subject: [PATCH 1457/2748] YARN-3554. Default value for maximum nodemanager
 connect wait time is too high. Contributed by
 Naganarasimha G R

(cherry picked from commit 9757864fd662b69445e0c600aedbe307a264982e)

Change-Id: Ib64f93962396ab2a9cddbddc65b056d5f8b8609d
---
 .../apache/hadoop/yarn/conf/YarnConfiguration.java |    2 +-
 .../src/main/resources/yarn-default.xml            |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/src/main/java/org/apache/hadoop/yarn/conf/YarnConfiguration.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/src/main/java/org/apache/hadoop/yarn/conf/YarnConfiguration.java
index c3d0e16..5d9d6ac 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/src/main/java/org/apache/hadoop/yarn/conf/YarnConfiguration.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/src/main/java/org/apache/hadoop/yarn/conf/YarnConfiguration.java
@@ -1596,7 +1596,7 @@ private static void addDeprecatedKeys() {
   public static final String CLIENT_NM_CONNECT_MAX_WAIT_MS =
       YARN_PREFIX + "client.nodemanager-connect.max-wait-ms";
   public static final long DEFAULT_CLIENT_NM_CONNECT_MAX_WAIT_MS =
-      15 * 60 * 1000;
+      3 * 60 * 1000;
 
   /** Time interval between each attempt to connect to NM */
   public static final String CLIENT_NM_CONNECT_RETRY_INTERVAL_MS =
diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/resources/yarn-default.xml b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/resources/yarn-default.xml
index 681b551..324c321 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/resources/yarn-default.xml
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/resources/yarn-default.xml
@@ -1158,7 +1158,7 @@
   <property>
     <description>Max time to wait to establish a connection to NM</description>
     <name>yarn.client.nodemanager-connect.max-wait-ms</name>
-    <value>900000</value>
+    <value>180000</value>
   </property>
 
   <property>
-- 
1.7.9.5

